src/ObjectUtils/PdbFileDia.cpp

```diff
@@ -84,47 +87,148 @@ ErrorMessageOr<void> PdbFileDia::LoadDataForPDB() {
+  UNDNAME_HAVE_PARAMETERS = 0x4000,       // The real template parameters are available.
+  UNDNAME_NO_ECSU = 0x8000,               // Suppresses enum/class/struct/union.
+  UNDNAME_NO_IDENT_CHAR_CHECK = 0x10000,  // Suppresses check for valid identifier characters.
+  UNDNAME_NO_PTR64 = 0x20000,             // Does not include ptr64 in output.
+};
+template <typename Consumer>
+ErrorMessageOr<void> ForEachSymbolWithSymTag(const enum SymTagEnum& sym_tag,
+                                             IDiaSymbol* dia_global_scope_symbol,
+                                             std::string_view file_path, Consumer&& consumer) {
   HRESULT result = 0;
   CComPtr<IDiaEnumSymbols> dia_enum_symbols;
-  result = dia_global_scope_symbol_->findChildren(SymTagFunction, NULL, nsNone, &dia_enum_symbols);
+  result = dia_global_scope_symbol->findChildren(sym_tag, NULL, nsNone, &dia_enum_symbols);
   if (result != S_OK) {
-    return ErrorMessage{
-        absl::StrFormat("findChildren failed for %s (%u)", file_path_.string(), result)};
+    return ErrorMessage{absl::StrFormat("findChildren failed for %s (%u)", file_path, result)};
   }
 
-  ModuleSymbols module_symbols;
   IDiaSymbol* dia_symbol = nullptr;
   ULONG celt = 0;
 
   while (SUCCEEDED(dia_enum_symbols->Next(1, &dia_symbol, &celt)) && (celt == 1)) {
     CComPtr<IDiaSymbol> dia_symbol_com_ptr = dia_symbol;
-    SymbolInfo symbol_info;
-
-    BSTR function_name = {};
-    if (dia_symbol->get_name(&function_name) != S_OK) continue;
-    symbol_info.set_demangled_name(orbit_base::ToStdString(function_name));
-    ErrorMessageOr<std::string> parameter_list_or_error = PdbDiaParameterListAsString(dia_symbol);
-    if (parameter_list_or_error.has_value()) {
-      symbol_info.set_demangled_name(symbol_info.demangled_name() +
-                                     parameter_list_or_error.value());
-    } else {
-      ORBIT_ERROR("Unable to retrieve parameter types of function %s. Error: %s",
-                  symbol_info.demangled_name(), parameter_list_or_error.error().message());
-    }
-    SysFreeString(function_name);
-
-    DWORD relative_virtual_address = 0;
-    if (dia_symbol->get_relativeVirtualAddress(&relative_virtual_address) != S_OK) continue;
-    symbol_info.set_address(relative_virtual_address + object_file_info_.load_bias);
-
-    ULONGLONG length = 0;
-    if (dia_symbol->get_length(&length) != S_OK) continue;
-    symbol_info.set_size(length);
-
-    *module_symbols.add_symbol_infos() = std::move(symbol_info);
+    std::invoke(std::forward<Consumer>(consumer), dia_symbol);
   }
 
+  return outcome::success();
+}
+}  // namespace
+
+ErrorMessageOr<orbit_grpc_protos::ModuleSymbols> PdbFileDia::LoadDebugSymbols() {
+  ModuleSymbols module_symbols;
+  absl::flat_hash_set<uint64_t> addresses_from_module_info_stream;
+
+  // Find the function symbols in the module info stream. For now, we ignore "blocks" and
+  // "thunks", as they don't add meaningful information.
```

Comment:
My feeling is, you have to explain the "for now" part and the "they don't add meaningful information" a little better. I imagine when reading this in a year it might be of little use.

Do I get this right: The documentation says there can be symbols hiding in thunks and blocks but we haven't observed that in any real-life PDB, that's why we are not considering them, right? If so I would write something along those lines.

Reply:
Good point. I rephrased the comment.